ipatests/test_integration/test_simple_replication.py

```diff
@@ -104,6 +105,24 @@ def test_ipa_custodia_check(self):
             [paths.IPA_CUSTODIA_CHECK, self.master.hostname]
         )
 
+    def test_fix_agreements(self):
+        """Test that upgrade fixes the list of attributes excluded from repl
+
+        Test for ticket 9385
+        """
+        # Prepare the server by removing some values from
+        # from the nsDS5ReplicatedAttributeList
+        segment = find_segment(self.master, self.replicas[0], "domain")
+        self.master.run_command([
+            "ipa", "topologysegment-mod", "domain", segment,
+            "--replattrs",
+            "(objectclass=*) $ EXCLUDE memberof idnssoaserial entryusn"])
+        # Run the upgrade
+        result = self.master.run_command(["ipa-server-upgrade"])
+        # Ensure that the uprgade updated the attribute
+        errmsg = "Error caught updating nsDS5ReplicatedAttributeList"
+        assert errmsg not in result.stdout_text
```

Comment:
I think we can re-generate the expected attribute output as too to also ensure that the value itself was set in LDAP despite no error being returned. I realize this is a bit on the pedantic side but it's more of an affirmative test that the attribute was set to what we expect.

Reply:
Thanks, the test change looks good.